### PR TITLE
Add integration tests for user defined local authenticator mgt.

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Task issue:
- https://github.com/wso2/product-is/issues/21216

Add integration tests to cover the management capability for user defined local authenticators.

Related PRs:
- https://github.com/wso2/carbon-identity-framework/pull/6172
- https://github.com/wso2/identity-api-server/pull/736

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws IOException, JSONException {
+
+        super.testInit(API_VERSION, swaggerDefinition, tenant);
+        testAuthenticatorConfig = createBaseUserDefinedLocalAuthenticator(
+                AuthenticatorPropertyConstants.AuthenticationType.IDENTIFICATION);
+        addUserDefinedAuthenticatorToIS();
+
```
Comment (malithie) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java:73]: Let's remove this line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED);
+
+        Response responseForDuplication = getResponseOfPost(AUTHENTICATOR_CUSTOM_API_BASE_PATH, body);
+        responseForDuplication.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("code", equalTo("AUT-60003"))
```
Comment (malithie) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java:244]: The same error code is duplicated

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+        updatePayload.getEndpoint().getAuthentication().getProperties().remove("username");
+        String body = UserDefinedLocalAuthenticatorPayload.convertToJasonPayload(updatePayload);
+        Response response = getResponseOfPutWithNoFilter(AUTHENTICATOR_CUSTOM_API_BASE_PATH + PATH_SEPARATOR +
+                customIdPId, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("code", equalTo("AUT-60003"))
```
Comment (malithie) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java:261]: The same error code is duplicated

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java (hunk 2 above)
Comment (Thisara-Welmilla) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java:243]: Will be fix with https://github.com/wso2/product-is/issues/22050

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java (hunk 3 above)
Comment (Thisara-Welmilla) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/management/v1/AuthenticatorFailureTest.java:260]: Will be fix with https://github.com/wso2/product-is/issues/22050

**Review by malithie — APPROVED** [review]

**Pull request merged**